File: src/Domain/Entities/Characters/CharacterCharacteristics.cs
namespace Crpg.Domain.Entities.Characters;

/// <summary>
/// Characteristics of a character.
/// </summary>
public class CharacterCharacteristics
{
    public CharacterAttributes Attributes { get; set; } = new();
    public CharacterSkills Skills { get; set; } = new();
    public CharacterWeaponProficiencies WeaponProficiencies { get; set; } = new();
}

File: src/WebUI/src/models/character-characteristics.ts
import CharacterAttributes from '@/models/character-attributes';
import CharacterSkills from '@/models/character-skills';
import CharacterWeaponProficiencies from '@/models/character-weapon-proficiencies';

export default interface CharacterCharacteristics {
  attributes: CharacterAttributes;
  skills: CharacterSkills;
  weaponProficiencies: CharacterWeaponProficiencies;
}
